engine/message/db.go

```diff
@@ -569,11 +575,37 @@ func (db *DB) _SendMessages(ctx context.Context, send SendFunc, status StatusFun
 		return errors.Wrap(err, "clear disabled status updates")
 	}
 
-	_, err = tx.Stmt(db.failDisabledCM).ExecContext(execCtx)
+	// processes disabled CMs and writes to alert log if disabled
+	cms, err := tx.Stmt(db.failDisabledCM).QueryContext(execCtx)
 	if err != nil {
 		return errors.Wrap(err, "check for disabled CMs")
 	}
 
+	type msgMeta struct {
+		MessageID string
+		AlertID   int
+	}
+
+	var msgs []msgMeta
+	for cms.Next() {
+		var msg msgMeta
+		err = cms.Scan(&msg.MessageID, &msg.AlertID)
+		if err != nil {
+			return errors.Wrap(err, "scan all disabled CM messages")
+		}
+		msgs = append(msgs, msg)
+	}
+	cms.Close()
+
+	for _, m := range msgs {
+		meta := alertlog.NotificationMetaData{
+			MessageID: m.MessageID,
+		}
+
+		// log failures
+		db.alertlogstore.MustLogTx(ctx, tx, m.AlertID, alertlog.TypeNotificationSent, meta)
```

Comment:
Add the user id to the context here -> `MustLogTx(permission.UserContext(ctx, userID, permission.RoleUser), tx, ...)`